web/src/components/graph/SystemGraph.tsx

```diff
@@ -0,0 +1,116 @@
+import { useTheme } from "@/context/theme-provider";
+import { Threshold } from "@/types/graph";
+import { useCallback, useEffect, useMemo } from "react";
+import Chart from "react-apexcharts";
+
+type SystemGraphProps = {
+  graphId: string;
+  name: string;
+  unit: string;
+  threshold: Threshold;
+  updateTimes: number[];
+  data: ApexAxisChartSeries;
+};
+export default function SystemGraph({
+  graphId,
+  name,
+  unit,
+  threshold,
+  updateTimes,
+  data,
+}: SystemGraphProps) {
+  const lastValue = useMemo<number>(
+    // @ts-expect-error y is valid
+    () => data[0].data[data[0].data.length - 1]?.y ?? 0,
+    [data],
+  );
+
+  const { theme, systemTheme } = useTheme();
+
+  const formatTime = useCallback(
+    (val: unknown) => {
+      const date = new Date(updateTimes[Math.round(val as number)] * 1000);
+      return `${date.getHours() > 12 ? date.getHours() - 12 : date.getHours()}:${date.getMinutes()}`;
```

Comment:
Could you do something like `return date.toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' });`? Currently zeros aren't padded so `7:05` looks like `7:5`.